wordpress-mobile/WordPress-iOS
# Editor session tracking

Adds the editor session tracking described in https://github.com/wordpress-mobile/gutenberg-mobile/issues/556

One change from the original spec is that I renamed the `reason` property to `outcome` in the `editor_session_end` event. The reason for this is that I noticed some undesired behavior when testing the first implementation: you could save a draft without leaving the editor, then closing it and it would register as a `cancel` (because it quit without pending changes).

After seeing that, I changed the logic so the editor session records save/publish events and forces those instead when sending `editor_session_end`. But sometimes those aren't the "reason" why the session ended anymore, "outcome" seems like a better description.

To test:

The best way to test the events is to look at the logs, but it gets too noisy with React Native. You can edit the Xcode scheme, and in Run > Arguments enable the `OS_ACTIVITY_MODE=disable` environment variable. You can also search for `🔵` in the log but that would filter out the event properties 😞 

Some things to test:

- Opening an editor should trigger the `editor_session_start` event
- Switching to classic/gutenberg/html should trigger the `editor_session_switch_editor` event, no `editor_session_start` when the other editor is created
- Saving/publishing shouldn't trigger the `editor_session_end` event. Nothing that doesn't close the editor should
- Closing the editor should trigger the `editor_session_end` event. The `outcome` should match whether the post has been published/saved, even if that was done before closing

cc @hypest
